frontend/src/components/Search.js

```diff
@@ -0,0 +1,147 @@
+import { useState, useEffect } from "react";
+//style
+import styled from "styled-components";
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+//redux
+import { useDispatch } from "react-redux";
+import { loadProductsData } from "../actions/productsAction";
+import { loadQuery } from "../actions/queryAction";
+
+const Search = () => {
+  //redux
+  const dispatch = useDispatch();
+  //state
+  const [input, setInput] = useState("");
+  const [price, setPrice] = useState("");
+  const [category, setCategory] = useState("");
+  const [query, setQuery] = useState("");
+
+  //event handlers
+  const handleInput = (e) => {
+    setInput(e.target.value);
+  };
+
+  const handlePriceClick = (e) => {
+    setPrice(e.target.value);
+  };
+
+  const handleCategoryClick = (e) => {
+    setCategory(e.target.value);
+  };
+
+  const handleSearchEvent = () => {
```

Comment:
I think this could be moved to the reducer (if you wanted to use `products.query`) in your Redux store.